asteroid/foo.py

```diff
@@ -0,0 +1,162 @@
+import os
+import torch
+import numpy as np
+import warnings
+
+try:
+    from typing import Protocol
+except ImportError:  # noqa
+    # Python < 3.8
+    class Protocol:
+        pass
+
+
+from asteroid.utils import get_device
+
+
+class Separatable(Protocol):
+    """Things that are separatable.
+
+    In addition to supporting the protocol specified below, implementations must
+    also work with `asteroid.utils.get_device` (eg. any `torch.nn.Module`).
+    """
+
+    def _separate(self, wav, **kwargs):
```

Comment:
Having one method start with `_` is a bit ugly but I didn't know how to change it without being backwards incompatible